src/server/app/api/controllers/userController.js

```diff
@@ -44,14 +44,25 @@ const userController = {
     }
   },
   async updateUser(req, h) {
-    let user;
-    let updatedUser;
+    let data, user, updatedUser;
     try {
-      if (!req.payload || !req.payload._id) throw new Error("Did not receive valid information for update.");
-      if (!ObjectId.isValid(req.payload._id)) throw new Error("User ID is not valid.");
-      user = await userStore.updateUser(req.payload);
+      if (!req.payload) throw new Error("Did not receive valid information for update.");
+      data = req.payload;
+
+      // Fetch user id from username
+      user = await userStore.getUserByPhoneOrEmail(data.user.username);
+      if (!ObjectId.isValid(user._id)) throw new Error("User ID is not valid.");
+
+      // Modify update data to remove username and add id
+      delete data["user"];
+      data["_id"] = ObjectId(user._id);
```

Comment:
Yeah if we could figure out a way to pass ID around instead of username, it feels like we could remove some complexity and error handling. And we wouldn't have to delete `data["user"]` either, if we `data` just had user id.

Reply:
Same as above-- I treated having to store the user ID on the client side as a potential security issue. I think I asked @thejoelpatrol about it but he could chime in on this if it's not that big a deal and I can take care of this complexity.

Reply:
> I treated having to store the user ID on the client side as a potential security issue

Like I said I above, I struggle to see how it could be a reasonable security issue. Everyone team I can think of uses user ID liberally to refer to users directly or via their association to other things, and are honestly much less likely to use `username` for that same purpose. Interested to hear Joel's take, though.

Reply:
We could consider encrypting user ID if it's a concern; some folks do that if they want to mitigate attacks around trying to make PATCH API updates using a user ID.

Reply:
Let me think about this again for a second and look around a little. Web APIs are not my specialty, though I do need to learn. At first brush, it seems sketchy to me to pass around internal identifiers from the DB to the user, but I have to think about what information is actually contained in that ID, other than them probably being sequential. I guess it reveals to the user a lower bound on how many users we have, for one thing.

Reply:
A solution I've seen to concerns around sequential ID's is using UUID for the DB identifier instead.

Reply:
Sorry for disappearing. This could work. If the goal is to be random, though, no need for a structured UUID, we can just use a random value. Read 16 bytes from /dev/urandom or the language equivalent.